nixos/modules/config/iproute2.nix

```diff
@@ -4,6 +4,17 @@ with lib;
 
 let
   cfg = config.networking.iproute2;
+
+  etc_iproute2 = pkgs.stdenv.mkDerivation {
+    name = "etc-iproute2";
+    src = pkgs.iproute2;
+    buildPhase = ''
+      mkdir $out
+      ln -s $src/etc/iproute2/* $out/
+      rm -f $out/rt_tables
+      echo ${lib.escapeShellArg cfg.rttablesExtraConfig} | cat $src/etc/iproute2/rt_tables - > $out/rt_tables
```

Comment:
The module works well on my machine.

But `echo ${lib.escapeShellArg cfg.rttablesExtraConfig}` might be a little bit wired here. Is this a reliable way to pass arbitrary nix string to a shell pipe? Or what is the standard way to append a nix string to a file in `stdenv`?

Reply:
What about

```bash
cat $src/etc/iproute2/rt_tables ${builtins.toFile "rt_tables-extra-config" cfg.rttablesExtraConfig} > $out/rt_tables
```

Reply:
That seems good! Didn't know about toFile, it's implemented and works fine in my local tests.